Gems/Atom/RPI/Code/Source/RPI.Reflect/Model/ModelAsset.cpp

```diff
@@ -261,13 +268,13 @@ namespace AZ
                         return false;
                     }
 
-                    const float* p = reinterpret_cast<const float*>(&positionRawBuffer[index0 * positionElementSize]);
+                    const float* p = &positionPtr[index0 * StepSize];
```

Comment:
any reason why you need a temporary variable.
```
a.Set(const_cast<float*>(&positionPtr[index0 * StepSize]));
```

Reply:
Suggestion: turn p into a value and remove the need for the const_cast on the set.

Reply:
Looks like it was done this way for speed
> // faster than AZ::Vector3 c-tor

Really Vector3::Set ought to take a const array

Reply:
I'll leave this as-is for now if that's cool, have the `p` pointer does make debugging easier too so it's useful for that (though perhaps having separate variables instead of reusing `p` might be nice...

Reply:
Also agreed `Vector3` should to take a const pointer/array, that seems like an oversight 🙁